exonum/src/storage/proof_list_index/proof.rs

```diff
@@ -17,7 +17,7 @@ use serde::ser::SerializeStruct;
 use serde::de::Error;
 use serde_json::{Error as SerdeJsonError, Value, from_value};
 
-use crypto::{Hash, hash};
+use crypto::{hash, CryptoHash, EntryHash};
```

Comment:
I seriously doubt `EntryHash` should be placed in the `crypto` module; it concerns only storage implementation details.